apps/web/src/pages/templates/workflow/WorkflowEditorPage.tsx

```diff
@@ -153,13 +175,50 @@ const WorkflowEditorPage = ({
                     >
                       Edit Template
                     </EditTemplateButton>
+                    <Button
+                      fullWidth
+                      mt={10}
+                      variant="outline"
+                      onClick={() => {
+                        setFilterOpen(true);
+                      }}
+                    >
+                      Add filter
+                    </Button>
                     <Divider my={30} />
                     {steps.map((i, index) => {
                       return index === activeStep ? (
                         <StepActiveSwitch key={index} index={activeStep} control={control} />
                       ) : null;
                     })}
                   </NavSection>
+                  <NavSection>
+                    {steps.map((i, index) => {
+                      if (index !== activeStep) {
+                        return null;
+                      }
+                      if (!i.filters || !Array.isArray(i.filters)) {
+                        return null;
+                      }
+
+                      return i.filters.map((group) => {
```

Comment:
Can we maybe also add some component tests here to make sure we cover all the cases?